src/ForkCMS/Utility/Thumbnails.php

```diff
@@ -157,4 +147,87 @@ public function getFolders(string $inPath, bool $includeSourceFolder = false): a
 
         return array_values($folders);
     }
+
+    /**
+     * Calculate the desired width based on the aspect ratio if needed
+     *
+     * @param int|null $desiredWidth
+     * @param int|null $desiredHeight
+     * @param int $currentWidth
+     * @param int $currentHeight
+     * @return int
+     */
+    private function calculateDesiredWidth(
+        ?int $desiredWidth,
+        ?int $desiredHeight,
+        int $currentWidth,
+        int $currentHeight
+    ): int {
+        if ($desiredWidth !== null) {
+            return $desiredWidth;
+        }
+
+        if ($desiredHeight === null) {
+            throw new \RuntimeException('A desired height is needed to calculate the desired width');
+        }
+
+        return $desiredHeight * ($currentWidth/$currentHeight);
+    }
+
+    /**
+     * Calculate the desired height based on the aspect ratio if needed
+     *
+     * @param int|null $desiredWidth
+     * @param int|null $desiredHeight
+     * @param $currentWidth
+     * @param $currentHeight
+     * @return int
+     */
+    private function calculateDesiredHeight(
+        ?int $desiredWidth,
+        ?int $desiredHeight,
+        int $currentWidth,
+        int $currentHeight
+    ): int {
+        if ($desiredHeight !== null) {
+            return $desiredHeight;
+        }
+
+        if ($desiredWidth === null) {
+            throw new \RuntimeException('A desired width is needed to calculate the desired height');
+        }
+
+        return $desiredWidth * ($currentHeight/$currentWidth);
+    }
+
+    private function calculateResizeBox(
+        int $width,
+        int $height,
+        ?int $desiredWidth,
+        ?int $desiredHeight
+    ): Box {
+        $desiredWidth = $this->calculateDesiredWidth($desiredWidth, $desiredHeight, $width, $height);
+        $desiredHeight = $this->calculateDesiredHeight($desiredWidth, $desiredHeight, $width, $height);
+
+        // if the current width and height are already in the desired width and height we don't need to do anything
+        if ($width >= $desiredWidth && $height >= $desiredHeight) {
+            return new Box($width, $height);
+        }
+
+        // handle width too small
+        if ($width < $desiredWidth) {
+            $scaleRatio = $desiredWidth/$width;
+            $height *= $scaleRatio;
+            $width *= $scaleRatio;
+        }
+
+        // handle height too small
+        if ($height < $desiredHeight) {
```

Comment:
What happens if the height and weight are to small? I probably am missing something, but in that case you could end up with  a resized image based on the height?

Reply:
That's why I didn't use an else
When asking these images for a crop of 300x300
![landscape](https://user-images.githubusercontent.com/3634654/81044481-b75ffc80-8eb4-11ea-9f05-971e20593ae0.png) 200x100
![portrait](https://user-images.githubusercontent.com/3634654/81044485-b7f89300-8eb4-11ea-9693-d0dc04ad726b.png) 100x200
![square](https://user-images.githubusercontent.com/3634654/81044488-b8912980-8eb4-11ea-9552-46927f1bd508.png) 200x200

Krijg je dit
![landscape](https://user-images.githubusercontent.com/3634654/81044567-e4acaa80-8eb4-11ea-861b-07129bfa4ca8.png)
![portrait](https://user-images.githubusercontent.com/3634654/81044569-e5454100-8eb4-11ea-8ddf-c70a230723e3.png)
![square](https://user-images.githubusercontent.com/3634654/81044570-e5ddd780-8eb4-11ea-927f-523011cb07a5.png)
Hij doet dus een upscale en dan een crop